frankDoc/src/test/java/nl/nn/adapterframework/frankdoc/AttributeTypeStrategyTest.java

```diff
@@ -3,151 +3,151 @@
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
 import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
 
+import nl.nn.adapterframework.frankdoc.doclet.FrankClassRepository;
+import nl.nn.adapterframework.frankdoc.doclet.TestUtil;
+import nl.nn.adapterframework.frankdoc.model.AttributeEnum;
 import nl.nn.adapterframework.frankdoc.model.AttributeType;
+import nl.nn.adapterframework.frankdoc.model.ElementChild;
+import nl.nn.adapterframework.frankdoc.model.FrankAttribute;
+import nl.nn.adapterframework.frankdoc.model.FrankDocModel;
 import nl.nn.adapterframework.util.XmlBuilder;
 
+@RunWith(Parameterized.class)
 public class AttributeTypeStrategyTest {
-	private String schemaString = getXsd();
-
-	@Test
-	public void whenNormalIntegerSuppliedThenAccepted() throws Exception {
-		validate(getIntTestXml("1"));
-	}
-
-	@Test
-	public void whenIntegerWithPlusSuppliedThenAccepted() throws Exception {
-		validate(getIntTestXml("+1"));
-	}
-
-	@Test
-	public void whenNegativeIntegerSuppliedThenAccepted() throws Exception {
-		validate(getIntTestXml("-12"));
-	}
-
-	@Test
-	public void whenRefSuppliedAsIntegerThenAccepted() throws Exception {
-		validate(getIntTestXml("${myVariable}"));
+	private String schemaStringAllowAttributeRef;
+	private String schemaStringAllowAttributeRefEnumValuesIgnoreCase;
+
+	@Parameters(name = "{0}")
+	public static Collection<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+			{"NormalInteger", getIntTestXml("1"), true, true},
+			{"IntegerWithPlus", getIntTestXml("+1"), true, true},
+			{"NegativeInteger", getIntTestXml("-12"), true, true},
+			{"RefAsInteger", getIntTestXml("${myVariable}"), true, true},
+			{"NonIntegerForInteger", getIntTestXml("aString"), false, false},
+			{"InvalidRefForInteger", getIntTestXml("${myVariable"), false, false},
+			{"True", getBoolTestXml("true"), true, true},
+			{"False", getBoolTestXml("false"), true, true},
+			{"RefAsBoolean", getBoolTestXml("${myVariable}"), true, true},
+			{"StringForBoolean", getBoolTestXml("aString"), false, false},
+			{"InvalidRefForBoolean", getBoolTestXml("${myVariable"), false, false},
+			{"AttributeActiveTrueLowerCase", getTestXmlActive("true"), true, true},
+			{"AttributeActiveFalseLowerCase", getTestXmlActive("false"), true, true},
+			{"AttributeActiveMixedCaseTrue", getTestXmlActive("True"), true, true},
+			{"AttributeActiveMixedCaseFalse", getTestXmlActive("False"), true, true},
+			{"RefAsAttributeActive", getTestXmlActive("${myVar}"), true, true},
+			{"AttributeActiveNotTrue", getTestXmlActive("!true"), true, true},
+			{"NegatedRefAsAttributeActive", getTestXmlActive("!${myVar}"), true, true},
+			{"StringForAttributeActive", getTestXmlActive("xxx"), false, false},
+			{"AttributeActiveConcatsMultipleValidValues", getTestXmlActive("${myVar}true"), false, false},
```

Comment:
Mijns inziens moet dit welk kunnen. Ik zie regelmatig dingen als `${baseurl.naar.endpoint}/collectie/test`.
Dit wordt gedaan om de url aan te kunnen passen op de verschillende DTAP stages zonder dat het path/querystring van de url veranderd.

Reply:
Voor attributes van het type string ben ik het met je eens. Dan moet bijvoorbeeld ${myVar}aap als waarde kunnen. Voor booleans denk ik dat ${myVar}aap niet moet kunnen. Bij booleans wil je nooit string concatenatie, denk ik.

Zal ik de code aanpassen voor type string attributes en boolean attributes ongemoeid laten?

Reply:
Attributes van type string zouden al goed moeten zijn. Als je een setter hebt van type string zonder dat er een enum getter is, dan krijg je in de XSD het type xs:string. Daar zouden variabele references al in moeten zitten.

Reply:
Misschien dat het nog handig is dit scenario prefix of suffix bij een string-property, ook te testen in deze klasse.

Reply:
Gedaan.